### PR TITLE
Fix UnwrittenWritableFieldAnalyzer with tests

### PR DESCRIPTION
**Pull request by genlu**
The bug is caused by not identifying increment/decrement operators as write operation.

@lgolding @srivatsn

### CODE REVIEW

File: src/Microsoft.DotNet.CodeFormatter.Analyzers/UnwrittenWritableFieldAnalyzer.cs [hunk 1]
```diff
@@ -44,6 +44,14 @@ public class UnwrittenWritableFieldAnalyzer : DiagnosticAnalyzer
                 SyntaxKind.SubtractAssignmentExpression
             };
 
+        private static readonly SyntaxKind[] s_unaryAssignmentKinds =
```
Comment (AlexGhiondea) [inline src/Microsoft.DotNet.CodeFormatter.Analyzers/UnwrittenWritableFieldAnalyzer.cs]: Could you have merged these with the compound operators?

Comment (genlu) [inline src/Microsoft.DotNet.CodeFormatter.Analyzers/UnwrittenWritableFieldAnalyzer.cs]: @AlexGhiondea Sure.

**Comment by AlexGhiondea** [comment]
LGTM

**Comment by srivatsn** [comment]
:+1:

**Pull request merged**